### DISCUSSION TITLE
Lesson 7, Part 19 - Interactions.s.sol

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#528] [General]
I am doing the lesson where we write scripts to interact with FundMe.sol and then write an interactions test to test fund and withdraw (see screencap of my terminal). I have the same code as Patrick but I'm getting errors that "fundMe" is an undeclared identifier. 

Edit: I added my code for Interactions.s.sol and Interactions.t.sol to a comment below.

Edit #2: I have solved the problem and here is what I had to do....first I had to install jq by running "sudo apt install jq" (to get Patrick's foundry-devops to work). Then I had to go to foundry.toml and add solc="0.8.19". Finally I had to run forge test --no-auto-detect instead of forge test. Then all the tests passed.

Here is my screen cap showing my InteractionsTest.t.sol code and the undeclared identifier error I am getting. Let me know if you need me to post any of the other code (eg the code for Interactions.s.sol):
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/138697105/2e750ef8-c7ea-4ea5-9ded-1f363d4655b6)

**Comment by shikhar229169** [comment]
Actually you don't have any state variable with the name fundMe.

So just add the state variable
```solidity
FundMe fundMe;
```

**Reply by 0xwtfwtf** [reply]
I fixed that but it still didn't work...Im gonna paste my code into a comment

**Reply by shikhar229169** [reply]
Can you paste the error that you are getting now.

**Reply by 0xwtfwtf** [reply]
It is the same errors shown in the screencap above about undeclared identifiers

**Comment by 0xwtfwtf** [comment]
Here is my code for Interactions.s.sol and Interactions.t.sol:

**Interactions.t.sol**

`//SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;

import {Test, console} from "forge-std/Test.sol";
import {FundMe} from "../../src/FundMe.sol";
import {DeployFundMe} from "../../script/DeployFundMe.s.sol"; 
import {FundFundMe, WithdrawFundMe} from "../../script/Interactions.s.sol";

contract InteractionsTest is Test {
    FundMe fundMe;

    address USER = makeAddr("user");
    uint256 constant SEND_VALUE = 10e17;
    uint256 constant STARTING_BALANCE = 10e19;

    function setUp() external {
        //fundMe = new FundMe(0x694AA1769357215DE4FAC081bf1f309aDC325306);
        DeployFundMe deploy = new DeployFundMe();
        fundMe = deploy.run();
        vm.deal(USER, STARTING_BALANCE);
    }

    function testUserCanFundInteractions() public {
        FundFundMe fundFundMe = new FundFundMe();
        fundFundMe.fundFundMe(address(fundMe));
    
        WithdrawFundMe withdrawFundMe = new WithdrawFundMe();
        withdrawFundMe.withdrawFundMe(address(fundMe));

        assert(address(fundMe).balance == 0);
    }
}`

**Interactions.s.sol:**

`// SPDX-License-Identifier: MIT

pragma solidity ^0.8.19;

import {Script, console} from "forge-std/Script.sol";
import {DevOpsTools} from "foundry-devops/src/DevOpsTools.sol";
import {FundMe} from "../src/FundMe.sol";

contract FundFundMe is Script {
    uint256 constant SEND_VALUE = 10e17;

    function fundFundMe(address mostRecentlyDeployed) public {
        vm.startBroadcast();
        FundMe(payable(mostRecentlyDeployed)).fund{value: SEND_VALUE}();
        vm.stopBroadcast();
        console.log("Funded FundMe with %s", SEND_VALUE);
    }

    function run() external {
        address mostRecentDeployed = DevOpsTools.get_most_recent_deployment(
            "FundMe",
            block.chainid
            );
        fundFundMe(mostRecentDeployed);
        
    }
}

contract WithdrawFundMe is Script {
    function withdrawFundMe(address mostRecentlyDeployed) public {
        vm.startBroadcast();
        FundMe(payable(mostRecentlyDeployed)).withdraw();
        vm.stopBroadcast();
    
    }
    function run() external {
        address mostRecentDeployed = DevOpsTools.get_most_recent_deployment(
            "FundMe",
            block.chainid
            );
        withdrawFundMe(mostRecentDeployed);
        }
}
//set up a script to interact with our contracts`

**Reply by theirrationalone** [reply]
Hey, I don't know why compiler showing that state variable "fundme" with small "m",

Please try to triage once again for me....

refactor: state variable like this...

```solidity
FundMe private s_fundMe;
```

after this 👆 rename all occurrences of "fundMe" and then **TRY AGAIN**

**Comment by JuanPedro0410** [comment]
i solved it changing this line: (fundMe, ) = deploy.run();
